terraform/modules/shared/lambda-function/main.tf

```diff
@@ -185,6 +187,11 @@ resource "aws_s3_bucket_notification" "bucket_notification" {
   depends_on = [aws_lambda_permission.allow_bucket_trigger]
 }
 
+resource "aws_lambda_function_event_invoke_config" "retry_config" {
+  function_name          = aws_lambda_function.function.function_name
```

Comment:
I'd be tempted to put a count in here so it's only created if retry_attempts isn't null: `count = var.retry_attempts != null ? 1 : 0`, just so it doesn't end up creating the resource for every lambda